cmd/kubeadm/app/util/version.go

```diff
@@ -78,6 +78,12 @@ func KubernetesReleaseVersion(version string) (string, error) {
 
 	// kubeReleaseLabelRegex matches labels such as: latest, latest-1, latest-1.10
 	if kubeReleaseLabelRegex.MatchString(versionLabel) {
+		var clientVersion string
+		// Check if a valid semantic client version can be obtained.
+		if _, err := versionutil.ParseSemantic(pkgversion.Get().String()); err == nil {
+			// ignore the error as kubeadmVersion() cannot fail at this point
+			clientVersion, _ = kubeadmVersion(pkgversion.Get().String())
+		}
```

Comment:
i'm not very happy with how this is turning out.
will revise tomorrow. the idea is to always get some sort of a client version: A) a valid one, B) empty string.

might as well remove the error check from `kubeadmVersion()` and adjust related sub-unit-tests.